runtime/jvmti/jvmtiFunctionTable.c

```diff
@@ -23,10 +23,32 @@
 #include "jvmtiHelpers.h"
 #include "jvmti_internal.h"
 
+#if JAVA_SPEC_VERSION >= 9
+#define JVMTI_MODULE_ENTRY(jvmtiGetAllModules) jvmtiGetAllModules
+#define JVMTI_MODULE_ENTRY(jvmtiGetNamedModule) jvmtiGetNamedModule
+#define JVMTI_MODULE_ENTRY(jvmtiAddModuleReads) jvmtiAddModuleReads
+#define JVMTI_MODULE_ENTRY(jvmtiAddModuleExports) jvmtiAddModuleExports
+#define JVMTI_MODULE_ENTRY(jvmtiAddModuleOpens) jvmtiAddModuleOpens
+#define JVMTI_MODULE_ENTRY(jvmtiAddModuleUses) jvmtiAddModuleUses
+#define JVMTI_MODULE_ENTRY(jvmtiAddModuleProvides) jvmtiAddModuleProvides
+#define JVMTI_MODULE_ENTRY(jvmtiIsModifiableModule) jvmtiIsModifiableModule
+#else /* JAVA_SPEC_VERSION >= 9 */
+#define JVMTI_MODULE_ENTRY(jvmtiGetAllModules) NULL
+#define JVMTI_MODULE_ENTRY(jvmtiGetNamedModule) NULL
+#define JVMTI_MODULE_ENTRY(jvmtiAddModuleReads) NULL
+#define JVMTI_MODULE_ENTRY(jvmtiAddModuleExports) NULL
+#define JVMTI_MODULE_ENTRY(jvmtiAddModuleOpens) NULL
+#define JVMTI_MODULE_ENTRY(jvmtiAddModuleUses) NULL
+#define JVMTI_MODULE_ENTRY(jvmtiAddModuleProvides) NULL
+#define JVMTI_MODULE_ENTRY(jvmtiIsModifiableModule) NULL
+#endif /* JAVA_SPEC_VERSION >= 9 */
```

Comment:
You should define `JVMTI_MODULE_ENTRY` exactly once:
```
#if JAVA_SPEC_VERSION >= 9
#  define JVMTI_MODULE_ENTRY(name) name
#else /* JAVA_SPEC_VERSION >= 9 */
#  define JVMTI_MODULE_ENTRY(name) NULL
#endif /* JAVA_SPEC_VERSION >= 9 */
```
The uses below are fine.